repository/client.repo.go

```diff
@@ -20,6 +24,39 @@ func NewClientRepository(c *context.Context, db *gorm.DB) *ClientRepository {
 	}
 }
 
+// List retrieves a list of clients based on provided pagination data and query.
+func (r *ClientRepository) List(
+	clients *[]*models.Client,
+	paginateData *models.PaginationData,
+	query insightInputs.ClientsQueryInput,
+) error {
+	dbTables := r.db.Model(&models.Client{})
+
+	return dbTables.Scopes(
+		helpers.Paginate(dbTables.Scopes(
+			r.nameLike(query.NameCont),
+		), paginateData),
+	).Order("id desc").Find(&clients).Error
+}
+
+// nameLike returns a function that filters client by name.
+func (r *ClientRepository) nameLike(nameLike *string) func(db *gorm.DB) *gorm.DB {
+	return func(db *gorm.DB) *gorm.DB {
+		if nameLike == nil {
+			return db
+		} else {
+			return db.Where(gorm.Expr(`lower(clients.name) LIKE ?`, strings.ToLower("%"+*nameLike+"%")))
+		}
+	}
+}
+
+// FindById finds a client by its ID.
+func (r *ClientRepository) FindById(client *models.Client, id int32) error {
+	dbTables := r.db.Model(&models.Client{})
+
+	return dbTables.First(&client, id).Error
+}
+
```

Comment:
những hàm find như này thì tham khảo bên khác, chỉ viết 1 hàm find duy nhất để find by attributes và mình dùng hàm này toàn bộ cho các loại tìm kiếm phụ thuộc attributes đầu vào thôi nhé